### ISSUE TITLE
Grid does not update select all state when data changes.

### ISSUE THREAD

**Issue by wholladay** [angular-ui/ui-grid#6033]
I have a ui-grid with multi-select enabled. Everything works fine, except when the data changes, the "select" column in the header does not properly show the state of the grid. I've created a [simple example in plunker](http://plnkr.co/edit/MTXR8Q?p=preview) that illustrates the issue. To reproduce:

1. Click on the "select" header element. Notice that all the rows are selected as they should be.
1. Click on the "Add Data" button. Notice that the new row is not selected, but the header is still showing a check mark as if all the rows are currently selected.

If you click on any of the rows at this point, the code is smart enough to properly set the header check value. It seems to me that as the data for the grid changes, the status of the "all selected" flag should be re-evaluated.

This sample code was created by forking the "[112 Add/Delete/Swap Data](http://ui-grid.info/docs/#/tutorial/112_swapping_data)" tutorial code and just adding multi-select to the grid.

**Comment by rescru** [comment]
Just had a brief look at the code and it seems the selection module is not listening for any changes in the underlying data. Therefore once the new row is added the selection model is outdated until a new selection takes place.

I have just recently started using this grid and its awesome!!

I will look to see if there is an event that can be listened to and then the selection model will need to listen to it.

Very keen to contribute a fix as I have found this grid extremely useful. I am however a bit of a newbie and not sure of the process to follow.

Anyone can point me in the right direction?
